src/server/fetch.ts

```diff
@@ -47,8 +47,7 @@ export async function playwrightFetch(context: BrowserContext, params: types.Fet
       agent = new HttpsProxyAgent(proxyOpts);
     }
 
-    // TODO(https://github.com/microsoft/playwright/issues/8381): set user agent
-    const fetchResponse = await sendRequest(context, new URL(params.url), {
+    const fetchResponse = await sendRequest(context, new URL(params.url, context._options.baseURL), {
```

Comment:
Let's use this function maybe? https://github.com/microsoft/playwright/blob/master/src/utils/utils.ts#L343

Reply:
It returns a string and swallows error, we need neither of those here.